e2e/test/iothub/device/MethodE2ECustomPayloadTests.cs

```diff
@@ -28,311 +28,125 @@ public class MethodE2ECustomPayloadTests : E2EMsTestBase
         private static readonly Dictionary<string, object> _dictResponse = new() { { "key1", new byte[] { 3, 5, 6 } }, { "key2", false } };
 
         private readonly string _devicePrefix = $"{nameof(MethodE2ETests)}_dev_";
-        private readonly string _modulePrefix = $"{nameof(MethodE2ETests)}_mod_";
         private const string MethodName = "MethodE2ETest";
 
-        private static readonly TimeSpan s_defaultMethodTimeoutMinutes = TimeSpan.FromMinutes(1);
+        private static readonly TimeSpan s_defaultMethodResponseTimeout = TimeSpan.FromMinutes(1);
 
+        // The deserialization code is the same irrespective of if the client was initialized over TCP or WS.
         [TestMethod]
         [Timeout(TestTimeoutMilliseconds)]
-        public async Task Method_DeviceReceivesMethodAndResponse_CustomPayload_Mqtt()
+        public async Task Method_DeviceReceivesMethodAndResponse_CustomType_MqttWs()
```

Comment:
I wanted to use parameterized tests instead but asserting the type was too complicated.